### PR TITLE
Remove strict_apply option, make the service account name configurable

### PR DESCRIPTION
**Pull request by ribbybibby**

### CODE REVIEW

**Comment by ffilippopoulos** [comment]
lgtm! Do you think we can add the rbac and SA manifests for the other namespaces as well under manifests?

File: README.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ kube-applier runs as a Pod in your cluster and watches the [Git repo](#mounting-
 
 Whenever a new commit to the repo occurs, or at a [specified interval](#run-interval), kube-applier performs a "run", issuing [kubectl apply](https://kubernetes.io/docs/user-guide/kubectl/v1.6/#apply) commands with pruning at namespace level. The convention is that level 1 subdirs of REPO_PATH represent k8s namespaces: the name of the dir is the same as the namespace and the dir contains manifests for the given namespace.
 
+kube-applier will try to fetch a Service Account under each namespace (as defined by the environment variable `SERVICE_ACCOUNT_NAME`) and will use it to run kubectl commands. It will error for all namespaces that do not contain a matching SA.
```
Comment (george-angel) [inline README.md]: `ACTING_SERVICE_ACCOUNT_NAME`?

**Comment by ribbybibby** [comment]
@ffilippopoulos if I understand kustomize, you mean adding a separate base for the other namespaces that does the RBAC and SA bits?

File: README.md (hunk 1 above)
Comment (ribbybibby) [reply README.md]: Hm, I suppose this disambiguates it from the SA of the actual kube-applier service. Although I think the distinction is clear enough without it.

I think I prefer the sound of `ASSUMED_SERVICE_ACCOUNT_NAME`, but I don't know if this is an accurate description of what is actually happening.

**Comment by ffilippopoulos** [comment]
@ribbybibby precisely, that was my idea. I don't know if it is good enough to be included here, just pitching it..

**Comment by george-angel** [comment]
I think its a good idea, then instruction can be to import the "main application" base where you want KA to run, and "managed namespaces" base for the SA/rbac.

**Comment by ribbybibby** [comment]
@george-angel @ffilippopoulos I've added a base for the managed namespaces. I moved the existing base under `manifests/main` and created a new folder: `manifests/managed-namespace`.

File: README.md (hunk 1 above)
Comment (george-angel) [reply README.md]: Happy for it to be something else, but if you have an application, and it has a flag called "service account name" - my first thought would be - the service account the application runs with, not the SA it gets from other namespaces to apply them.

Comment (ribbybibby) [reply README.md]: Changed.

File: manifests/auth/example/auth-kube-applier-patch.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: kube-applier-assumed
+  namespace: managed-namespace-name
```
Comment (george-angel) [inline manifests/auth/example/auth-kube-applier-patch.yaml]: I would remove these

File: manifests/auth/example/auth-kube-applier-patch.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (hectorhuertas) [inline manifests/auth/example/auth-kube-applier-patch.yaml]: I'd remove this

File: manifests/auth/base/auth-kube-applier.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (hectorhuertas) [inline manifests/auth/base/auth-kube-applier.yaml]: I'd remove this

File: manifests/auth/base/auth-kube-applier.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: kube-applier-assumed
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: kube-applier-sa-get
```
Comment (hectorhuertas) [inline manifests/auth/base/auth-kube-applier.yaml]: This could be called just `kube-applier`?

Comment (ribbybibby) [reply manifests/auth/base/auth-kube-applier.yaml]: I quite like the fact that it isn't just `kube-applier` - there's too much 'kube-applier' over all the objects here, ha. 

What about `kube-applier-role`? Leaves it open to be appended to or amended.

Comment (george-angel) [reply manifests/auth/base/auth-kube-applier.yaml]: Its already a "role" - its like having a variable called "error_count_variable"

Comment (hectorhuertas) [reply manifests/auth/base/auth-kube-applier.yaml]: There is a lot of `kube-applier` around, but that's common in kube ecosystem. If there aren't any other KA roles, I'd lend over it being called just `kube-applier`

Comment (ribbybibby) [reply manifests/auth/base/auth-kube-applier.yaml]: `kube-applier` it is

**Pull request merged**